openai_batch/create_batch_input.py

```diff
@@ -0,0 +1,162 @@
+"""
+Construct an input file from prompts
+"""
+
+import argparse
+import json
+import typing
+
+import openai
+from openai.types.chat import ChatCompletionUserMessageParam
+from openai.types.chat.completion_create_params import CompletionCreateParamsNonStreaming
+from openai.types import EmbeddingCreateParams
+
+from .providers import _add_provider_arg, get_provider_by_model
+
+
+def get_parser():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "input",
+        nargs="?",
+        default="-",
+        type=argparse.FileType("r"),
+        help="Filename or '-' for stdin. One prompt per line.",
+    )
+
+    parser.add_argument(
+        "output",
+        nargs="?",
+        default="-",
+        type=argparse.FileType("w", encoding="utf-8"),
+        help="Filename or '-' for stdout. This will be the batch input .jsonl file.",
+    )
+
+    parser.add_argument(
+        "--model",
+        default="gpt-4",
+        help="Which model to target.",
+    )
+
+    parser.add_argument(
+        "--embedding",
+        "-e",
+        help="Whether this is an embedding model",
+        default=False,
+        action="store_true",
+    )
+
+    # Need to know batch input file size limits, this can vary by provider.
+    _add_provider_arg(parser)
+
+    return parser
+
+
+class BatchType:
+    CHAT_COMPLETION = "chat_completion"
+    EMBEDDING = "embedding"
+
+
+class Batch:
+    def __init__(self, output_file, custom_id_prefix="line"):
+        self.output_file = output_file
+        self.custom_id_prefix = custom_id_prefix
+        self.n_bytes = 0
+        self.n_requests = 0
+        self.model = None
+        self.provider = None
```

Comment:
Add `provider` to args so it can be explicitly specified to skip the auto detection.